packages/build-tools/src/steps/functions/repack.ts

```diff
@@ -0,0 +1,174 @@
+        id: 'platform',
+        required: false,
+        allowedValueTypeName: BuildStepInputValueTypeName.STRING,
+      }),
+      BuildStepInput.createProvider({
+        id: 'output_path',
+        allowedValueTypeName: BuildStepInputValueTypeName.STRING,
+        required: false,
+      }),
+      BuildStepInput.createProvider({
+        id: 'embed_bundle_assets',
+        allowedValueTypeName: BuildStepInputValueTypeName.BOOLEAN,
+        required: false,
+      }),
+    ],
+    outputProviders: [
+      BuildStepOutput.createProvider({
+        id: 'output_path',
+        required: true,
+      }),
+    ],
+    fn: async (stepsCtx, { inputs, outputs, env }) => {
+      const projectRoot = stepsCtx.workingDirectory;
+      const verbose = stepsCtx.global.env['EAS_VERBOSE'] === '1';
+
+      const platform =
+        (inputs.platform.value as Platform) ?? stepsCtx.global.staticContext.job.platform;
+      if (![Platform.ANDROID, Platform.IOS].includes(platform)) {
+        throw new Error(
+          `Unsupported platform: ${platform}. Platform must be "${Platform.ANDROID}" or "${Platform.IOS}"`
+        );
+      }
+
+      const repackLogger = createBunyanLoggerAdapter(stepsCtx.logger);
+      const repackSpawnAsync = createSpawnAsyncStepAdapter(verbose, stepsCtx.logger);
+
+      const tmpDir = await fs.promises.mkdtemp(path.join(os.tmpdir(), `repack-`));
+      const workingDirectory = path.join(tmpDir, 'working-directory');
+      await fs.promises.mkdir(workingDirectory, { recursive: true });
+      stepsCtx.logger.info(`Created temporary workingDirectory: ${workingDirectory}`);
+
+      const sourceAppPath = inputs.source_app_path.value as string;
+      const outputPath =
+        (inputs.output_path.value as string) ??
+        path.join(tmpDir, `repacked${path.extname(sourceAppPath)}`);
+
+      stepsCtx.logger.info('Repacking the app...');
+      if (platform === Platform.IOS) {
+        await repackAppIosAsync({
+          platform: 'ios',
+          projectRoot,
+          sourceAppPath,
+          outputPath,
+          workingDirectory,
+          // TODO: add iosSigningOptions
+          logger: repackLogger,
+          spawnAsync: repackSpawnAsync,
+          verbose,
+          env: {
+            FASTLANE_DISABLE_COLORS: '1',
+            FASTLANE_SKIP_UPDATE_CHECK: '1',
+            SKIP_SLOW_FASTLANE_WARNING: 'true',
+            FASTLANE_HIDE_TIMESTAMP: 'true',
+            LC_ALL: 'en_US.UTF-8',
+            ...env,
+          },
+        });
+      } else if (platform === Platform.ANDROID) {
+        await repackAppAndroidAsync({
+          platform: 'android',
+          projectRoot,
+          sourceAppPath,
+          outputPath,
+          workingDirectory,
+          // TODO: add androidSigningOptions
+          logger: repackLogger,
+          spawnAsync: repackSpawnAsync,
+          verbose,
+          env,
+        });
+      } else {
+        throw new Error('Unsupported platform');
```

Comment:
You might want to `switch (platform)` and make type assumptions about `platform`. We already validated `platform` in 63-66.